Source/plugins/IShell.h

```diff
@@ -255,66 +273,82 @@ namespace PluginHost {
         virtual Core::hresult Hibernate(const uint32_t timeout) = 0;
         virtual reason Reason() const = 0;
 
+        virtual void Register(IConnectionServer::INotification* sink) = 0;
+        virtual void Unregister(const IConnectionServer::INotification* sink) = 0;
+
         // Method to access, in the main process space, the channel factory to submit JSON objects to be send.
         // This method will return a error if it is NOT in the main process.
         /* @stubgen:stub */
         virtual uint32_t Submit(const uint32_t Id, const Core::ProxyType<Core::JSON::IElement>& response) = 0;
 
-        // Method to access, in the main space, a COM factory to instantiate objects out-of-process.
-        // This method will return a nullptr if it is NOT in the main process.
-        /* @stubgen:stub */
-        virtual ICOMLink* COMLink() = 0;
-
         inline void Register(RPC::IRemoteConnection::INotification* sink)
         {
-            ICOMLink* handler(COMLink());
+            ASSERT(sink != nullptr);
```

Comment:
Given the remark above ;-) Maybe we should also move this to a deditcated interface and than QueryInterface<IRemoteConnection>() and than Register ? What do you think?